build_debian.sh

```diff
@@ -618,6 +619,56 @@ then
 
 fi
 
+# #################
+#   secure boot 
+# #################
+if [[ $SECURE_UPGRADE_MODE == 'dev' || $SECURE_UPGRADE_MODE == "prod" && $SONIC_ENABLE_SECUREBOOT_SIGNATURE != 'y' ]]; then
+    # note: SONIC_ENABLE_SECUREBOOT_SIGNATURE is a feature that signing just kernel, 
```

Comment:
I think we should club this now and single solution for UEFI secure boot, We added this earlier to sign the Linux kernel and copy the signed boot loader component (Ex:  SHIM and Monolithic GRUB) from HW vendor ONIE ESP partition.

Reply:
could be OK, but we need to check that there are no conflicts,
are you planning to use this flag in other places also?

Reply:
It may be better to call/restate Prod and Dev as signing types instead of stating/giving the perspective of Dev signing expects keys from local and production signing expects remote server. Not all vendor has remote signing server that does not mean their images are not production image. HLD can provide guidelines on what is the preferable method in case of the release signing key to keep securely. I see this infra supports both local and remote key signing methods but the definition of the workflow may require to be restated.

Reply:
This is a naming conflict as I see (not a functional issue).
I will do some additional review of the naming conventions and see if from the security POV it's OK to named it local and remote instead dev and prod.
@Yarden-Z 
@eddyk-nvidia